test/suite/test_prepare_discover06.py

```diff
@@ -0,0 +1,197 @@
+
+        # Advance stable timestamp past the prepare timestamp
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(150))
+
+        # Create checkpoint to persist the prepared transaction
+        checkpoint_session = self.conn.open_session()
+        checkpoint_session.checkpoint()
+        checkpoint_session.close()
+
+        # Phase 3: Reopen as follower and pickup checkpoint
+        self.pr("=== Phase 3: Reopen as follower and pickup checkpoint ===")
+
+        # Get the checkpoint metadata before closing
+        checkpoint_meta = self.disagg_get_complete_checkpoint_meta()
+
+        # Configure as follower with checkpoint pickup (not using backup)
+        follower_config = self.conn_base_config + \
+                         'disaggregated=(role="follower",' + \
+                         f'checkpoint_meta="{checkpoint_meta}")'
+
+        # Reopen connection as follower
+        self.reopen_conn(config=follower_config)
+
+        # Verify committed data is visible but prepared data is not
+        cursor = self.session.open_cursor(self.uri)
+
+        # Should see committed data at timestamp 60
+        self.session.begin_transaction("read_timestamp=" + self.timestamp_str(60))
+        self.assertEqual(cursor[1], "committed_value_1")
+        self.assertEqual(cursor[2], "committed_value_2")
+        self.assertEqual(cursor[3], "committed_value_3")
+
+        # Prepared data should not be visible yet
+        cursor.set_key(4)
+        self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+        cursor.set_key(5)
+        self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+        cursor.set_key(6)
+        self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+        self.session.rollback_transaction()
+
+        cursor.close()
+
+        # Open prepared discover cursor to find pending prepared transactions
+        prepared_discover_cursor = self.session.open_cursor("prepared_discover:")
+
+        # Walk through prepared discover cursor to find our transaction
+        discover_session = self.conn.open_session()
+        count = 0
+        found_prepared_id = None
+
+        while prepared_discover_cursor.next() == 0:
+            count += 1
+            prepared_id = prepared_discover_cursor.get_key()
+            self.assertEqual(prepared_id, 123)  # Should find our prepared transaction
+            found_prepared_id = prepared_id
+
+            # Claim the prepared transaction
+            discover_session.begin_transaction("claim_prepared_id=" + self.prepared_id_str(prepared_id))
+            break
+
+        self.assertEqual(count, 1)  # Should find exactly one prepared transaction
+        self.assertEqual(found_prepared_id, 123)
+        prepared_discover_cursor.close()
+
+        # Commit the claimed prepared transaction at timestamp 200
+        discover_session.commit_transaction("commit_timestamp=" + self.timestamp_str(200) +
```

Comment:
expecting it to fail here since we can't modify on stable table, but this test is passing on this line